doc/protocol.md

```diff
@@ -674,6 +674,193 @@ sent by the server might have the following additional fields:
 
 No payload messages will be sent by this channel.
 
+Payload: metrics1
+-----------------
+
+With this payload type, you can monitor a large number of metrics,
+such as all of the metrics exposed by PCP, the Performance Copilot.
+
+In addition to monitoring live values of the metrics, you can also
+access archives of previously recorded values.
+
+You specify which metrics to monitor in the options of the "open"
+command for a new channel.  If some of the metrics are not available,
+or their attributes do not match what you have specified, the channel
+will close with an error.
+
+One of the options is the "source" of the metrics.  Many details of
+how to specify metrics and how the channel will work in detail depend
+on the type of the source, and are explained in specific sections
+below.
+
+The general open options are:
+
+ * "source" (string): The source of the metrics:
+
+   * "direct": PCP metrics from plugins that are loaded into the
+     Cockpit bridge directly.  Use this when in doubt.
+
+   * "pcmd": PCP metrics from the local PCP daemon.
+
+ * "metrics" (array): Descriptions of the metrics to use.  The exact
+   format and semantics depend on the source.  See the specific
+   sections below.
+
+ * "instances" (array of strings, optional): When specified, only the
+   listed instances are included in the reported samples.
+
+ * "omit_instances" (array of strings, optional): When specified, the
+   listed instances are omitted from the reported samples.  Only one
+   of "instances" and "omit_instances" can be specified.
```

Comment:
All the other options use dashes rather than underscores. I think this should stay consistent.